src/Core/Geographic/Coordinates.js

```diff
@@ -1,441 +1,250 @@
+        return target.set(this.x, this.y, this.z);
     }
 
-    extent.dimensions(dimension);
+    /**
+     * Returns coordinates in the wanted [CRS]{@link http://inspire.ec.europa.eu/theme/rs}.
+     *
+     * @param {string} crs - The CRS to convert the Coordinates into.
+     * @param {Coordinates} [target] - The target to put the converted
+     * Coordinates into. If not specified a new one will be created.
+     *
+     * @return {Coordinates} - The resulting Coordinates after the conversion.
+     *
+     * @example
+     * const position = { longitude: 2.33, latitude: 48.24, altitude: 24999549 };
+     * const coords = new Coordinates('EPSG:4326', position.longitude, position.latitude, position.altitude); // Geographic system
+     * const coordinates = coords.as('EPSG:4978'); // Geocentric system
+     *
+     * @example
+     * const position = { x: 20885167, y: 849862, z: 23385912 };
+     * const coords = new Coordinates('EPSG:4978', position.x, position.y, position.z);  // Geocentric system
+     * const coordinates = coords.as('EPSG:4326');  // Geographic system
+     *
+     * @example
+     * new Coordinates('EPSG:4326', longitude: 2.33, latitude: 48.24, altitude: 24999549).as('EPSG:4978'); // Geocentric system
+     *
+     * @example
+     * new Coordinates('EPSG:4978', x: 20885167, y: 849862, z: 23385912).as('EPSG:4326'); // Geographic system
+     */
+    as(crs, target = new Coordinates(crs)) {
+        if (this.crs == crs) {
+            target.copy(this);
+            // there is a bug for converting anything from and to 4978 with proj4
+            // https://github.com/proj4js/proj4js/issues/195
+            // the workaround is to use an intermediate projection, like EPSG:4326
+        } else if (crs == 'EPSG:4978') {
+            if (CRS.is4326(this.crs)) {
+                ellipsoid.cartographicToCartesian(this, vectorPivot);
+                target.set(vectorPivot);
+            } else {
+                coordPivot.set(proj4cache(this.crs, 'EPSG:4326').forward([this.x, this.y]), this.z);
+                coordPivot.as('EPSG:4978', target);
+            }
+        } else if (this.crs == 'EPSG:4978') {
+            if (CRS.is4326(crs)) {
+                ellipsoid.cartesianToCartographic(this, target);
+            } else {
+                this.as('EPSG:4326', coordPivot);
+                target.set(proj4cache('EPSG:4326', crs).forward([coordPivot.x, coordPivot.y]), coordPivot.z);
+            }
+        } else if (CRS.is4326(this.crs) && crs == 'EPSG:3857') {
+            this.y = THREE.Math.clamp(this.y, -89.999999, 89.999999);
+            target.set(proj4cache(this.crs, crs).forward([this.x, this.y]), this.z);
+        } else {
+            target.set(proj4cache(this.crs, crs).forward([this.x, this.y]), this.z);
+        }
 
-    const x = CRS.isGeographic(this.crs) ? this.longitude() : this.x();
-    const y = CRS.isGeographic(this.crs) ? this.latitude() : this.y();
+        target.crs = crs;
```

Comment:
it will simplify the code if you write `setCrs()` function.

```js
// this will replace
coord.crs = crs;
coord.set(...);

// by 
coord.set(...).setCrs(crs);
```

Reply:
Ok, but do you see `CRS.isValid` called in `setCrs` ?

Reply:
not necessarily